tests_e2e/orchestrator/lib/templates/ext_seq_vmss_template.json

```diff
@@ -0,0 +1,321 @@
+                            }
+                        }
+                    }
+                ],
+                "backendAddressPools": [
+                    {
+                        "name": "[variables('bepoolName')]"
+                    }
+                ],
+                "inboundNatPools": [
+                    {
+                        "name": "[variables('natpoolName')]",
+                        "properties": {
+                            "FrontendIPConfiguration": {
+                                "Id": "[variables('feIpConfigId')]"
+                            },
+                            "BackendPort": 22,
+                            "Protocol": "tcp",
+                            "FrontendPortRangeStart": 3500,
+                            "FrontendPortRangeEnd": 4500
+                        }
+                    }
+                ],
+                "loadBalancingRules": [
+                    {
+                        "name": "ProbeRule",
+                        "properties": {
+                            "frontendIPConfiguration": {
+                                "id": "[variables('feIpConfigId')]"
+                            },
+                            "backendAddressPool": {
+                                "id": "[variables('bepoolID')]"
+                            },
+                            "protocol": "Tcp",
+                            "frontendPort": 80,
+                            "backendPort": 80,
+                            "idleTimeoutInMinutes": 5,
+                            "probe": {
+                                "id": "[variables('sshProbeId')]"
+                            }
+                        }
+                    }
+                ],
+                "probes": [
+                    {
+                        "name": "[variables('sshProbeName')]",
+                        "properties": {
+                            "protocol": "tcp",
+                            "port": 22,
+                            "intervalInSeconds": 5,
+                            "numberOfProbes": 2
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "apiVersion": "2023-03-01",
+            "type": "Microsoft.Compute/virtualMachineScaleSets",
+            "name": "[parameters('vmName')]",
+            "location": "[resourceGroup().location]",
+            "dependsOn": [
+                "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]",
+                "[concat('Microsoft.Network/loadBalancers/', variables('lbName'))]"
+            ],
+            "sku": {
+                "name": "Standard_D2s_v3",
+                "tier": "Standard",
+                "capacity": 3
+            },
+            "properties": {
+                "orchestrationMode": "Flexible",
+                "virtualMachineProfile": {
+                    "extensionProfile": {
+                        "extensions": [
```

Comment:
This template includes extensions and dependencies which will be added to the scale set at creation.

This is different from the current DCR scenario. The current DCR scenario creates the VMSS, and then repeatedly updates the extensions/dependencies on the scale set in the test to validate the order in which the extensions were enabled. This can be done because the DCR scenario creates a VMSS with 'Uniform' orchestration mode. With Uniform orchestration mode you can update the scale set model and each instance in the scale set will be updated with the new model (including extensions). 

We cannot use 'Uniform' orchestration mode for the VMSS in this scenario. Lisa requires that we use 'Flexible' orchestration mode, because while setting up the test environment, LISA looks for the VM in the resource group. In flexible orchestration mode, a VM for each instance is created in the resource group, but for uniform mode there are no VM resources created for the instances. 

As a result, this test will look a little different from DCR. We must add the extensions/dependencies at VMSS creation so that each instance will get the extensions/dependencies. We can only validate once, because we cannot update the VM instances after VMSS creation. 

Pavan mentioned Flexible VMSS will support UpgradePolicy in the next 1-2 semesters. So I added a TODO to improve this scenario when that happens